### PR TITLE
Add support for ranger random forest models in lime.

### PR DESCRIPTION
**Pull request by nielsenmarkus11**

### CODE REVIEW

File: R/models.R [hunk 1]
```diff
@@ -190,6 +190,19 @@ predict_model.H2OModel <- function(x, newdata, type, ...){
...
 }
+#' @export
+predict_model.ranger <- function(x, newdata, type, ...) {
+  if (!requireNamespace('ranger', quietly = TRUE)) {
+    stop('The ranger package is required for predicting ranger models')
+  }
+  res <- predict(x, data = newdata, ...)
+  switch(
+    type,
+    raw = as.data.frame(Response = colnames(res$predictions)[max.col(res$predictions)], stringsAsFactors = FALSE),
```
Comment (thomasp85) [inline R/models.R]: `raw` is used for regression models to return a single column data.frame with the predicted response. This seems to return a class membership? Or maybe I don't understand rangers return format enough..?

Comment (nielsenmarkus11) [reply R/models.R]: Is the single column of the class prediction not what is wanted?  I patterned mine after the predict_model.lda.  ranger doesn't seem to have a built in class prediction, so I just picked the class with the highest probability.

Comment (thomasp85) [reply R/models.R]: Raw is not used by classifiers at all - only regressors

Comment (nielsenmarkus11) [reply R/models.R]: Oh... makes sense. I'll make some adjustments.  Thanks!

**Comment by thomasp85** [comment]
There are some small fixes, but I think it is easier to just fix this after the merge

**Comment by thomasp85** [comment]
Thanks

**Comment by nielsenmarkus11** [comment]
NP. :)

**Pull request merged**